### ISSUE TITLE
Make adding the various plumbing for new bindings easier.

### ISSUE THREAD

**Issue by peterhuene** [peterhuene/azure-functions-rs#170]
Each new Azure Functions binding needs to define two binding-related types:

* A binding type in `azure-functions-shared` that is responsible for JSON serialization of the binding into `function.json`.
* A binding type in `azure-functions-codegen` that is responsible for parsing the `#[binding]` attribute on an Azure Function and emitting a declaration of the related binding type from `azure-functions-shared`.

This is a lot of copy-and-paste boilerplate code that is a great candidate for macro-ization.

I propose the following:

* Add a procedural macro to `azure-functions-shared-codegen` that can be used to decorate a binding type in `auzre-functions-shared`.  It will be responsible for generating a `serde_json::Serialize` implementation for the binding type.
* Add a procedural macro to `azure-functions-shared-codegen` that can be used to decorate a binding type in `azure-functions-codegen`.  It will be responsible for generating a `TryFrom<AttributeArguments>` and a `quote::ToTokens` implementation for the binding type.

**Comment by peterhuene** [comment]
Also, revisit the crate structure.  It may be possible to merge some of this stuff together.

**Comment by peterhuene** [comment]
I ended up removing the duplicated binding definitions as well.  Originally they were separate because the ones in the `azure-functions-codgen` crate depended on `proc_macro` types. 

I've extracted these out and based them upon `proc_macro2` now.  This enables us to have a single type the is responsible for both serializing the bindings to JSON, parsing the attributes, and converting the binding to tokens for macro output.
